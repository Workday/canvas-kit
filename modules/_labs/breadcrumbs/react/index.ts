import Breadcrumbs from './lib/Breadcrumbs';

export default Breadcrumbs;
export {Breadcrumbs};
export * from './lib/Breadcrumbs';
