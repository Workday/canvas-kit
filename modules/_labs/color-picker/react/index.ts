export {default, default as ColorPicker, ColorPickerProps} from './lib/ColorPicker';
export {Swatch} from './lib/parts/Swatch';
