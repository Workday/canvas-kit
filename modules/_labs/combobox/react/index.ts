import Combobox from './lib/Combobox';

export default Combobox;
export {Combobox};
export * from './lib/Combobox';
