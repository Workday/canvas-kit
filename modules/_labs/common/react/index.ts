export * from './lib/Flex';
export * from './lib/Box';
