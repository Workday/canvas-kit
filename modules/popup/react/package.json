{
  "name": "@workday/canvas-kit-react-popup",
  "version": "4.0.0-beta.1",
  "description": "Popup to display some content on top of another",
  "author": "Workday, Inc. (https://www.workday.com)",
  "license": "Apache-2.0",
  "main": "dist/commonjs/index.js",
  "module": "dist/es6/index.js",
  "sideEffects": false,
  "types": "dist/es6/index.d.ts",
  "repository": {
    "type": "git",
    "url": "http://github.com/Workday/canvas-kit/tree/master/modules/popup/react"
  },
  "files": [
    "dist/",
    "lib/",
    "index.ts",
    "ts3.5/**/*"
  ],
  "typesVersions": {
    "<=3.5": {
      "*": [
        "ts3.5/*"
      ]
    }
  },
  "scripts": {
    "watch": "yarn build:es6 -w",
    "test": "echo \"Error: no test specified\" && exit 1",
    "clean": "rimraf dist && rimraf .build-info && mkdirp dist",
    "build:cjs": "tsc -p tsconfig.cjs.json",
    "build:es6": "tsc -p tsconfig.es6.json",
    "build:rebuild": "npm-run-all clean build",
    "build:downlevel-dts": "yarn run downlevel-dts dist ts3.5/dist",
    "build": "npm-run-all --parallel build:cjs build:es6 --sequential build:downlevel-dts",
    "depcheck": "node ../../../utils/check-dependencies-exist.js"
  },
  "keywords": [
    "canvas",
    "canvas-kit",
    "react",
    "components",
    "workday",
    "popup"
  ],
  "dependencies": {
    "@emotion/core": "^10.0.28",
    "@emotion/is-prop-valid": "^0.8.2",
    "@emotion/styled": "^10.0.27",
    "@workday/canvas-kit-react-button": "^4.0.0-beta.1",
    "@workday/canvas-kit-react-card": "^4.0.0-beta.1",
    "@workday/canvas-kit-react-common": "^4.0.0-beta.1",
    "@workday/canvas-kit-react-core": "^4.0.0-beta.1",
    "@workday/canvas-system-icons-web": "^1.0.20",
    "uuid": "^3.3.3"
  },
  "peerDependencies": {
    "react": ">= 16.8 < 17"
  }
}
