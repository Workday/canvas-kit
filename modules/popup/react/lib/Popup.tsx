import * as React from 'react';
import Card from '@workday/canvas-kit-react-card';
import styled from 'react-emotion';
import {IconButton, ButtonSizes} from '@workday/canvas-kit-react-button';
import {CanvasDepthValue, spacing} from '@workday/canvas-kit-react-core';
import {TransformOrigin, getTranslateFromOrigin} from '@workday/canvas-kit-react-common';
import {xIcon} from '@workday/canvas-system-icons-web';
import {keyframes} from 'emotion';

export enum PopupPadding {
  zero = '0px',
  s = '16px',
  l = '32px',
}

export interface PopupProps {
  padding: PopupPadding;
  transformOrigin: TransformOrigin;
  closeIconSize: ButtonSizes.Small | ButtonSizes.Medium;
  popupRef?: React.Ref<HTMLDivElement>;
  handleClose?: () => void;
  width?: number | string;
  heading?: React.ReactNode;
  depth?: CanvasDepthValue;
}

const closeIconSpacing = spacing.xs;
const closeIconSpacingSmall = 10;

const popupAnimation = (transformOrigin: TransformOrigin) => {
  const translate = getTranslateFromOrigin(transformOrigin, spacing.xxs);

  return keyframes`
    0% {
      opacity: 0;
      transform: translate(${translate.x}px, ${translate.y}px);
    }
    100% {
      opacity: 1;
      transform: translate(0);
    }
  `;
};

const Container = styled('div')<Pick<PopupProps, 'transformOrigin' | 'width'>>(
  {
    position: 'relative',
  },
  ({width}) => width && {width},
  ({transformOrigin}) => ({
    animation: popupAnimation(transformOrigin),
    animationDuration: '150ms',
    animationTimingFunction: 'ease-out',
    transformOrigin: `${transformOrigin.vertical} ${transformOrigin.horizontal}`,
  })
);

const CloseIconContainer = styled('div')<Pick<PopupProps, 'closeIconSize'>>(
  {
    position: 'absolute',
  },
  ({closeIconSize}) => ({
    right: closeIconSize === ButtonSizes.Small ? closeIconSpacingSmall : closeIconSpacing,
    top: closeIconSize === ButtonSizes.Small ? closeIconSpacingSmall : closeIconSpacing,
  })
);

export default class Popup extends React.Component<PopupProps> {
  static Padding = PopupPadding;

  static defaultProps = {
    padding: Popup.Padding.l,
    closeIconSize: ButtonSizes.Medium,
    transformOrigin: {
      horizontal: 'center',
      vertical: 'top',
    },
  };

  public render() {
    const {
      handleClose,
      padding,
      width,
      heading,
      depth,
      closeIconSize,
      transformOrigin,
      popupRef,
      ...elemProps
    } = this.props;
    return (
      <Container
        transformOrigin={transformOrigin}
        width={width}
        role="dialog"
        innerRef={popupRef}
        {...elemProps}
      >
        {handleClose && (
          <CloseIconContainer closeIconSize={closeIconSize}>
            <IconButton
              buttonType={IconButton.Types.Plain}
              buttonSize={closeIconSize}
              onClick={handleClose}
              icon={xIcon}
              title="Close"
              aria-label="Close"
            />
          </CloseIconContainer>
        )}
        <Card depth={depth} heading={heading} width={width} padding={padding}>
          {this.props.children}
        </Card>
      </Container>
    );
  }
}
