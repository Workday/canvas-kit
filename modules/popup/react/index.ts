import Popup from './lib/Popup';

export default Popup;
export {Popup};
export * from './lib/Popup';
export * from './lib/Popper';
export * from './lib/useCloseOnEscape';
export * from './lib/useFocusTrap';
export * from './lib/useCloseOnOutsideClick';
export * from './lib/usePopupStack';
export * from './lib/useAssistiveHideSiblings';
export * from './lib/useBringToTopOnClick';
export * from './lib/getTransformFromPlacement';
