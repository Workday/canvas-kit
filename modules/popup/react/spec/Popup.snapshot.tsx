import * as React from 'react';
import Popup from '../lib/Popup';
import * as renderer from 'react-test-renderer';
import {ButtonSizes} from '@workday/canvas-kit-react';

describe('Popup Snapshots', () => {
  test('renders as expected', () => {
    const component = renderer.create(<Popup padding={Popup.Padding.s} />);
    expect(component).toMatchSnapshot();
  });
  test('renders Popup with close icon', () => {
    const component = renderer.create(<Popup handleClose={jest.fn()} padding={Popup.Padding.s} />);
    expect(component).toMatchSnapshot();
  });
  test('renders Popup with different padding', () => {
    const component = renderer.create(<Popup handleClose={jest.fn()} padding={Popup.Padding.l} />);
    expect(component).toMatchSnapshot();
  });
  test('renders Popup with children elements', () => {
    const component = renderer.create(
      <Popup handleClose={jest.fn()} padding={Popup.Padding.zero}>
        <span>hello world</span>
      </Popup>
    );
    expect(component).toMatchSnapshot();
  });
  test('renders Popup without close icon', () => {
    const component = renderer.create(
      <Popup padding={Popup.Padding.zero}>
        <span>hello world</span>
      </Popup>
    );
    expect(component).toMatchSnapshot();
  });
  test('renders Popup with transformOrigin', () => {
    const component = renderer.create(
      <Popup transformOrigin={{vertical: 'top', horizontal: 'center'}} padding={Popup.Padding.zero}>
        <span>hello world</span>
      </Popup>
    );
    expect(component).toMatchSnapshot();
  });

  test('renders Popup with a smaller close icon', () => {
    const component = renderer.create(
      <Popup
        closeIconSize={ButtonSizes.Small}
        transformOrigin={{vertical: 'top', horizontal: 'center'}}
        padding={Popup.Padding.zero}
      >
        <span>hello world</span>
      </Popup>
    );
    expect(component).toMatchSnapshot();
  });
});
