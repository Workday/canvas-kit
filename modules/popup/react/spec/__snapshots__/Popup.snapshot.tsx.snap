// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popup Snapshots renders Popup with a smaller close icon 1`] = `
.emotion-2 {
  position: relative;
  -webkit-animation: animation-1htt9yb;
  animation: animation-1htt9yb;
  -webkit-animation-duration: 150ms;
  animation-duration: 150ms;
  -webkit-animation-timing-function: ease-out;
  animation-timing-function: ease-out;
  -webkit-transform-origin: top center;
  -ms-transform-origin: top center;
  transform-origin: top center;
}

.emotion-0 {
  font-size: 14px;
  line-height: 20px;
  font-weight: 400;
  color: #494949;
  font-family: "Roboto","Helvetica Neue","Helvetica",Arial,sans-serif;
}

.emotion-1 {
  background-color: #ffffff;
  border: 1px solid #ced3d9;
  border-radius: 8px;
  box-sizing: border-box;
  box-shadow: 0px 4px 8px 0 rgba(0,0,0,0.1);
  padding: 0px;
}

<div
  className="emotion-2 emotion-3"
  role="dialog"
>
  <div
    className="emotion-1"
  >
    <div
      className="emotion-0"
    >
      <span>
        hello world
      </span>
    </div>
  </div>
</div>
`;

exports[`Popup Snapshots renders Popup with children elements 1`] = `
.emotion-7 {
  position: relative;
  -webkit-animation: animation-1htt9yb;
  animation: animation-1htt9yb;
  -webkit-animation-duration: 150ms;
  animation-duration: 150ms;
  -webkit-animation-timing-function: ease-out;
  animation-timing-function: ease-out;
  -webkit-transform-origin: top center;
  -ms-transform-origin: top center;
  transform-origin: top center;
}

.emotion-5 {
  font-size: 14px;
  line-height: 20px;
  font-weight: 400;
  color: #494949;
  font-family: "Roboto","Helvetica Neue","Helvetica",Arial,sans-serif;
}

.emotion-3 {
  position: absolute;
  right: 12px;
  top: 12px;
}

.emotion-1 {
  box-sizing: border-box;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: 13px;
  border-radius: 999px;
  border: 1px solid transparent;
  box-shadow: none;
  position: relative;
  cursor: pointer;
  outline: none;
  -webkit-transition: all 120ms linear;
  transition: all 120ms linear;
  border-width: 0px;
  background-color: transparent;
  margin: -8px;
  width: 40px;
  height: 40px;
}

.emotion-1:hover:active {
  -webkit-transition-duration: 40ms;
  transition-duration: 40ms;
}

.emotion-1:disabled,
.emotion-1:disabled:active {
  cursor: default;
  box-shadow: none;
}

.emotion-1 .wd-icon {
  display: inline-block;
  vertical-align: middle;
}

.emotion-1 span .wd-icon-fill,
.emotion-1 span .wd-icon-accent {
  -webkit-transition: fill 120ms ease-in;
  transition: fill 120ms ease-in;
  fill: #7b858f;
}

.emotion-1:focus {
  background-color: transparent;
}

.emotion-1:focus span .wd-icon-fill,
.emotion-1:focus span .wd-icon-accent {
  fill: #7b858f;
}

.emotion-1:hover:focus {
  background-color: transparent;
}

.emotion-1:active,
.emotion-1:focus:active,
.emotion-1:hover:active {
  background-color: transparent;
  border-color: transparent;
}

.emotion-1:active span .wd-icon-fill,
.emotion-1:focus:active span .wd-icon-fill,
.emotion-1:hover:active span .wd-icon-fill,
.emotion-1:active span .wd-icon-accent,
.emotion-1:focus:active span .wd-icon-accent,
.emotion-1:hover:active span .wd-icon-accent {
  fill: #333d47;
}

.emotion-1:hover {
  background-color: transparent;
}

.emotion-1:hover span .wd-icon-fill,
.emotion-1:hover span .wd-icon-accent {
  fill: #333d47;
}

.emotion-1:disabled,
.emotion-1:active:disabled,
.emotion-1:focus:disabled,
.emotion-1:hover:disabled {
  background-color: transparent;
}

.emotion-1:disabled span .wd-icon-fill,
.emotion-1:active:disabled span .wd-icon-fill,
.emotion-1:focus:disabled span .wd-icon-fill,
.emotion-1:hover:disabled span .wd-icon-fill,
.emotion-1:disabled span .wd-icon-accent,
.emotion-1:active:disabled span .wd-icon-accent,
.emotion-1:focus:disabled span .wd-icon-accent,
.emotion-1:hover:disabled span .wd-icon-accent {
  fill: #b9c0c7;
}

.emotion-1:not([disabled]):focus {
  -webkit-animation: animation-os2b9x 100ms;
  animation: animation-os2b9x 100ms;
  box-shadow: 0 0 0 0px #ffffff,0 0 0 2px #0875e1;
}

.emotion-1:not([disabled]):active {
  border-color: transparent;
  -webkit-animation: animation-os2b9x 100ms;
  animation: animation-os2b9x 100ms;
  box-shadow: 0 0 0 0px #ffffff,0 0 0 2px #0875e1;
}

.emotion-0 {
  display: inline-block;
}

.emotion-0 svg {
  display: block;
}

.emotion-0 .wd-icon-fill {
  fill: #7b858f;
}

.emotion-0:hover .wd-icon-fill {
  fill: #333d47;
}

.emotion-0 .wd-icon-accent {
  fill: #7b858f;
}

.emotion-0:hover .wd-icon-accent {
  fill: #333d47;
}

.emotion-0 .wd-icon-background {
  fill: transparent;
}

.emotion-0:hover .wd-icon-background {
  fill: transparent;
}

.emotion-6 {
  background-color: #ffffff;
  border: 1px solid #ced3d9;
  border-radius: 8px;
  box-sizing: border-box;
  box-shadow: 0px 4px 8px 0 rgba(0,0,0,0.1);
  padding: 0px;
}

<div
  className="emotion-7 emotion-8"
  role="dialog"
>
  <div
    className="emotion-3 emotion-4"
  >
    <button
      aria-label="Close"
      className="emotion-1 emotion-2"
      onClick={[MockFunction]}
      title="Close"
    >
      <span
        className="emotion-0"
        dangerouslySetInnerHTML={
          Object {
            "__html": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"24\\" height=\\"24\\" class=\\"wd-icon-x wd-icon\\" focusable=\\"false\\" role=\\"presentation\\" viewBox=\\"0 0 24 24\\"><g fill-rule=\\"evenodd\\" class=\\"wd-icon-container\\"><path d=\\"M5.996 4.582a.5.5 0 0 0-.717-.003l-.7.7a.504.504 0 0 0 .003.717L10.586 12l-6.004 6.004a.5.5 0 0 0-.003.717l.7.7c.197.197.514.2.717-.003L12 13.414l6.004 6.004a.5.5 0 0 0 .717.003l.7-.7a.504.504 0 0 0-.003-.717L13.414 12l6.004-6.004a.5.5 0 0 0 .003-.717l-.7-.7a.504.504 0 0 0-.717.003L12 10.586 5.996 4.582z\\" class=\\"wd-icon-fill\\"/></g></svg>",
          }
        }
      />
    </button>
  </div>
  <div
    className="emotion-6"
  >
    <div
      className="emotion-5"
    >
      <span>
        hello world
      </span>
    </div>
  </div>
</div>
`;

exports[`Popup Snapshots renders Popup with close icon 1`] = `
.emotion-7 {
  position: relative;
  -webkit-animation: animation-1htt9yb;
  animation: animation-1htt9yb;
  -webkit-animation-duration: 150ms;
  animation-duration: 150ms;
  -webkit-animation-timing-function: ease-out;
  animation-timing-function: ease-out;
  -webkit-transform-origin: top center;
  -ms-transform-origin: top center;
  transform-origin: top center;
}

.emotion-6 {
  background-color: #ffffff;
  border: 1px solid #ced3d9;
  border-radius: 8px;
  box-sizing: border-box;
  box-shadow: 0px 4px 8px 0 rgba(0,0,0,0.1);
  padding: 16px;
}

.emotion-5 {
  font-size: 14px;
  line-height: 20px;
  font-weight: 400;
  color: #494949;
  font-family: "Roboto","Helvetica Neue","Helvetica",Arial,sans-serif;
}

.emotion-3 {
  position: absolute;
  right: 12px;
  top: 12px;
}

.emotion-1 {
  box-sizing: border-box;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: 13px;
  border-radius: 999px;
  border: 1px solid transparent;
  box-shadow: none;
  position: relative;
  cursor: pointer;
  outline: none;
  -webkit-transition: all 120ms linear;
  transition: all 120ms linear;
  border-width: 0px;
  background-color: transparent;
  margin: -8px;
  width: 40px;
  height: 40px;
}

.emotion-1:hover:active {
  -webkit-transition-duration: 40ms;
  transition-duration: 40ms;
}

.emotion-1:disabled,
.emotion-1:disabled:active {
  cursor: default;
  box-shadow: none;
}

.emotion-1 .wd-icon {
  display: inline-block;
  vertical-align: middle;
}

.emotion-1 span .wd-icon-fill,
.emotion-1 span .wd-icon-accent {
  -webkit-transition: fill 120ms ease-in;
  transition: fill 120ms ease-in;
  fill: #7b858f;
}

.emotion-1:focus {
  background-color: transparent;
}

.emotion-1:focus span .wd-icon-fill,
.emotion-1:focus span .wd-icon-accent {
  fill: #7b858f;
}

.emotion-1:hover:focus {
  background-color: transparent;
}

.emotion-1:active,
.emotion-1:focus:active,
.emotion-1:hover:active {
  background-color: transparent;
  border-color: transparent;
}

.emotion-1:active span .wd-icon-fill,
.emotion-1:focus:active span .wd-icon-fill,
.emotion-1:hover:active span .wd-icon-fill,
.emotion-1:active span .wd-icon-accent,
.emotion-1:focus:active span .wd-icon-accent,
.emotion-1:hover:active span .wd-icon-accent {
  fill: #333d47;
}

.emotion-1:hover {
  background-color: transparent;
}

.emotion-1:hover span .wd-icon-fill,
.emotion-1:hover span .wd-icon-accent {
  fill: #333d47;
}

.emotion-1:disabled,
.emotion-1:active:disabled,
.emotion-1:focus:disabled,
.emotion-1:hover:disabled {
  background-color: transparent;
}

.emotion-1:disabled span .wd-icon-fill,
.emotion-1:active:disabled span .wd-icon-fill,
.emotion-1:focus:disabled span .wd-icon-fill,
.emotion-1:hover:disabled span .wd-icon-fill,
.emotion-1:disabled span .wd-icon-accent,
.emotion-1:active:disabled span .wd-icon-accent,
.emotion-1:focus:disabled span .wd-icon-accent,
.emotion-1:hover:disabled span .wd-icon-accent {
  fill: #b9c0c7;
}

.emotion-1:not([disabled]):focus {
  -webkit-animation: animation-os2b9x 100ms;
  animation: animation-os2b9x 100ms;
  box-shadow: 0 0 0 0px #ffffff,0 0 0 2px #0875e1;
}

.emotion-1:not([disabled]):active {
  border-color: transparent;
  -webkit-animation: animation-os2b9x 100ms;
  animation: animation-os2b9x 100ms;
  box-shadow: 0 0 0 0px #ffffff,0 0 0 2px #0875e1;
}

.emotion-0 {
  display: inline-block;
}

.emotion-0 svg {
  display: block;
}

.emotion-0 .wd-icon-fill {
  fill: #7b858f;
}

.emotion-0:hover .wd-icon-fill {
  fill: #333d47;
}

.emotion-0 .wd-icon-accent {
  fill: #7b858f;
}

.emotion-0:hover .wd-icon-accent {
  fill: #333d47;
}

.emotion-0 .wd-icon-background {
  fill: transparent;
}

.emotion-0:hover .wd-icon-background {
  fill: transparent;
}

<div
  className="emotion-7 emotion-8"
  role="dialog"
>
  <div
    className="emotion-3 emotion-4"
  >
    <button
      aria-label="Close"
      className="emotion-1 emotion-2"
      onClick={[MockFunction]}
      title="Close"
    >
      <span
        className="emotion-0"
        dangerouslySetInnerHTML={
          Object {
            "__html": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"24\\" height=\\"24\\" class=\\"wd-icon-x wd-icon\\" focusable=\\"false\\" role=\\"presentation\\" viewBox=\\"0 0 24 24\\"><g fill-rule=\\"evenodd\\" class=\\"wd-icon-container\\"><path d=\\"M5.996 4.582a.5.5 0 0 0-.717-.003l-.7.7a.504.504 0 0 0 .003.717L10.586 12l-6.004 6.004a.5.5 0 0 0-.003.717l.7.7c.197.197.514.2.717-.003L12 13.414l6.004 6.004a.5.5 0 0 0 .717.003l.7-.7a.504.504 0 0 0-.003-.717L13.414 12l6.004-6.004a.5.5 0 0 0 .003-.717l-.7-.7a.504.504 0 0 0-.717.003L12 10.586 5.996 4.582z\\" class=\\"wd-icon-fill\\"/></g></svg>",
          }
        }
      />
    </button>
  </div>
  <div
    className="emotion-6"
  >
    <div
      className="emotion-5"
    />
  </div>
</div>
`;

exports[`Popup Snapshots renders Popup with different padding 1`] = `
.emotion-7 {
  position: relative;
  -webkit-animation: animation-1htt9yb;
  animation: animation-1htt9yb;
  -webkit-animation-duration: 150ms;
  animation-duration: 150ms;
  -webkit-animation-timing-function: ease-out;
  animation-timing-function: ease-out;
  -webkit-transform-origin: top center;
  -ms-transform-origin: top center;
  transform-origin: top center;
}

.emotion-5 {
  font-size: 14px;
  line-height: 20px;
  font-weight: 400;
  color: #494949;
  font-family: "Roboto","Helvetica Neue","Helvetica",Arial,sans-serif;
}

.emotion-3 {
  position: absolute;
  right: 12px;
  top: 12px;
}

.emotion-1 {
  box-sizing: border-box;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: 13px;
  border-radius: 999px;
  border: 1px solid transparent;
  box-shadow: none;
  position: relative;
  cursor: pointer;
  outline: none;
  -webkit-transition: all 120ms linear;
  transition: all 120ms linear;
  border-width: 0px;
  background-color: transparent;
  margin: -8px;
  width: 40px;
  height: 40px;
}

.emotion-1:hover:active {
  -webkit-transition-duration: 40ms;
  transition-duration: 40ms;
}

.emotion-1:disabled,
.emotion-1:disabled:active {
  cursor: default;
  box-shadow: none;
}

.emotion-1 .wd-icon {
  display: inline-block;
  vertical-align: middle;
}

.emotion-1 span .wd-icon-fill,
.emotion-1 span .wd-icon-accent {
  -webkit-transition: fill 120ms ease-in;
  transition: fill 120ms ease-in;
  fill: #7b858f;
}

.emotion-1:focus {
  background-color: transparent;
}

.emotion-1:focus span .wd-icon-fill,
.emotion-1:focus span .wd-icon-accent {
  fill: #7b858f;
}

.emotion-1:hover:focus {
  background-color: transparent;
}

.emotion-1:active,
.emotion-1:focus:active,
.emotion-1:hover:active {
  background-color: transparent;
  border-color: transparent;
}

.emotion-1:active span .wd-icon-fill,
.emotion-1:focus:active span .wd-icon-fill,
.emotion-1:hover:active span .wd-icon-fill,
.emotion-1:active span .wd-icon-accent,
.emotion-1:focus:active span .wd-icon-accent,
.emotion-1:hover:active span .wd-icon-accent {
  fill: #333d47;
}

.emotion-1:hover {
  background-color: transparent;
}

.emotion-1:hover span .wd-icon-fill,
.emotion-1:hover span .wd-icon-accent {
  fill: #333d47;
}

.emotion-1:disabled,
.emotion-1:active:disabled,
.emotion-1:focus:disabled,
.emotion-1:hover:disabled {
  background-color: transparent;
}

.emotion-1:disabled span .wd-icon-fill,
.emotion-1:active:disabled span .wd-icon-fill,
.emotion-1:focus:disabled span .wd-icon-fill,
.emotion-1:hover:disabled span .wd-icon-fill,
.emotion-1:disabled span .wd-icon-accent,
.emotion-1:active:disabled span .wd-icon-accent,
.emotion-1:focus:disabled span .wd-icon-accent,
.emotion-1:hover:disabled span .wd-icon-accent {
  fill: #b9c0c7;
}

.emotion-1:not([disabled]):focus {
  -webkit-animation: animation-os2b9x 100ms;
  animation: animation-os2b9x 100ms;
  box-shadow: 0 0 0 0px #ffffff,0 0 0 2px #0875e1;
}

.emotion-1:not([disabled]):active {
  border-color: transparent;
  -webkit-animation: animation-os2b9x 100ms;
  animation: animation-os2b9x 100ms;
  box-shadow: 0 0 0 0px #ffffff,0 0 0 2px #0875e1;
}

.emotion-0 {
  display: inline-block;
}

.emotion-0 svg {
  display: block;
}

.emotion-0 .wd-icon-fill {
  fill: #7b858f;
}

.emotion-0:hover .wd-icon-fill {
  fill: #333d47;
}

.emotion-0 .wd-icon-accent {
  fill: #7b858f;
}

.emotion-0:hover .wd-icon-accent {
  fill: #333d47;
}

.emotion-0 .wd-icon-background {
  fill: transparent;
}

.emotion-0:hover .wd-icon-background {
  fill: transparent;
}

.emotion-6 {
  background-color: #ffffff;
  border: 1px solid #ced3d9;
  border-radius: 8px;
  box-sizing: border-box;
  box-shadow: 0px 4px 8px 0 rgba(0,0,0,0.1);
  padding: 32px;
}

<div
  className="emotion-7 emotion-8"
  role="dialog"
>
  <div
    className="emotion-3 emotion-4"
  >
    <button
      aria-label="Close"
      className="emotion-1 emotion-2"
      onClick={[MockFunction]}
      title="Close"
    >
      <span
        className="emotion-0"
        dangerouslySetInnerHTML={
          Object {
            "__html": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"24\\" height=\\"24\\" class=\\"wd-icon-x wd-icon\\" focusable=\\"false\\" role=\\"presentation\\" viewBox=\\"0 0 24 24\\"><g fill-rule=\\"evenodd\\" class=\\"wd-icon-container\\"><path d=\\"M5.996 4.582a.5.5 0 0 0-.717-.003l-.7.7a.504.504 0 0 0 .003.717L10.586 12l-6.004 6.004a.5.5 0 0 0-.003.717l.7.7c.197.197.514.2.717-.003L12 13.414l6.004 6.004a.5.5 0 0 0 .717.003l.7-.7a.504.504 0 0 0-.003-.717L13.414 12l6.004-6.004a.5.5 0 0 0 .003-.717l-.7-.7a.504.504 0 0 0-.717.003L12 10.586 5.996 4.582z\\" class=\\"wd-icon-fill\\"/></g></svg>",
          }
        }
      />
    </button>
  </div>
  <div
    className="emotion-6"
  >
    <div
      className="emotion-5"
    />
  </div>
</div>
`;

exports[`Popup Snapshots renders Popup with transformOrigin 1`] = `
.emotion-2 {
  position: relative;
  -webkit-animation: animation-1htt9yb;
  animation: animation-1htt9yb;
  -webkit-animation-duration: 150ms;
  animation-duration: 150ms;
  -webkit-animation-timing-function: ease-out;
  animation-timing-function: ease-out;
  -webkit-transform-origin: top center;
  -ms-transform-origin: top center;
  transform-origin: top center;
}

.emotion-0 {
  font-size: 14px;
  line-height: 20px;
  font-weight: 400;
  color: #494949;
  font-family: "Roboto","Helvetica Neue","Helvetica",Arial,sans-serif;
}

.emotion-1 {
  background-color: #ffffff;
  border: 1px solid #ced3d9;
  border-radius: 8px;
  box-sizing: border-box;
  box-shadow: 0px 4px 8px 0 rgba(0,0,0,0.1);
  padding: 0px;
}

<div
  className="emotion-2 emotion-3"
  role="dialog"
>
  <div
    className="emotion-1"
  >
    <div
      className="emotion-0"
    >
      <span>
        hello world
      </span>
    </div>
  </div>
</div>
`;

exports[`Popup Snapshots renders Popup without close icon 1`] = `
.emotion-2 {
  position: relative;
  -webkit-animation: animation-1htt9yb;
  animation: animation-1htt9yb;
  -webkit-animation-duration: 150ms;
  animation-duration: 150ms;
  -webkit-animation-timing-function: ease-out;
  animation-timing-function: ease-out;
  -webkit-transform-origin: top center;
  -ms-transform-origin: top center;
  transform-origin: top center;
}

.emotion-0 {
  font-size: 14px;
  line-height: 20px;
  font-weight: 400;
  color: #494949;
  font-family: "Roboto","Helvetica Neue","Helvetica",Arial,sans-serif;
}

.emotion-1 {
  background-color: #ffffff;
  border: 1px solid #ced3d9;
  border-radius: 8px;
  box-sizing: border-box;
  box-shadow: 0px 4px 8px 0 rgba(0,0,0,0.1);
  padding: 0px;
}

<div
  className="emotion-2 emotion-3"
  role="dialog"
>
  <div
    className="emotion-1"
  >
    <div
      className="emotion-0"
    >
      <span>
        hello world
      </span>
    </div>
  </div>
</div>
`;

exports[`Popup Snapshots renders as expected 1`] = `
.emotion-2 {
  position: relative;
  -webkit-animation: animation-1htt9yb;
  animation: animation-1htt9yb;
  -webkit-animation-duration: 150ms;
  animation-duration: 150ms;
  -webkit-animation-timing-function: ease-out;
  animation-timing-function: ease-out;
  -webkit-transform-origin: top center;
  -ms-transform-origin: top center;
  transform-origin: top center;
}

.emotion-1 {
  background-color: #ffffff;
  border: 1px solid #ced3d9;
  border-radius: 8px;
  box-sizing: border-box;
  box-shadow: 0px 4px 8px 0 rgba(0,0,0,0.1);
  padding: 16px;
}

.emotion-0 {
  font-size: 14px;
  line-height: 20px;
  font-weight: 400;
  color: #494949;
  font-family: "Roboto","Helvetica Neue","Helvetica",Arial,sans-serif;
}

<div
  className="emotion-2 emotion-3"
  role="dialog"
>
  <div
    className="emotion-1"
  >
    <div
      className="emotion-0"
    />
  </div>
</div>
`;
